### PR TITLE
fix: Exclude model_extra from serialization to prevent secret leakage

### PR DESCRIPTION
**Pull request by NodeJSmith**
AppManifest and HassetteConfig use extra="allow", so unrecognised env vars / TOML keys silently land in model_extra and were included in model_dump() output — potentially exposing secrets like tokens.

Add an ExcludeExtrasMixin that overrides model_dump and model_dump_json to strip model_extra keys by default.  Callers can still opt-in via an explicit `include=` argument.

## Summary

What change are you making and why?

## Changes

- Key points of the change

## Impact

- Any breaking changes, migrations, or follow-ups to note?

## Testing

- [ ] Not run yet
- [ ] Tests pass (`uv run nox -s tests`)
- [ ] Manual check done

## Related

Fixes #
Relates to #

## Notes

Screenshots, logs, or anything else reviewers should see.

### CODE REVIEW

File: src/hassette/config/classes.py [hunk 1]
```diff
@@ -48,7 +48,36 @@ def __init__(self, settings_cls: type[BaseSettings], toml_file: PathType | None
...
+        """Serialize declared fields only; extra fields are excluded for privacy."""
+        extra_keys = self._get_extra_keys()
+        if extra_keys and "include" not in kwargs:
+            exclude = (exclude or set()) | extra_keys
+        return super().model_dump(exclude=exclude, **kwargs)  # type: ignore[misc]
+
+    def model_dump_json(self, *, exclude: set[str] | None = None, **kwargs: Any) -> str:
+        """Serialize declared fields only; extra fields are excluded for privacy."""
+        extra_keys = self._get_extra_keys()
+        if extra_keys and "include" not in kwargs:
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/hassette/config/classes.py]: Resolved — changed the check to `kwargs.get("include") is not None` in both `model_dump` and `model_dump_json`. Added tests for the `include=None` edge case.
